bokeh/models/tools.py

```diff
@@ -413,3 +413,13 @@ class HoverTool(Tool):
         the ``Rect`` and ``Quad`` glyphs are supported.
 
     """)
+
+    mode = Enum("point", "hline", "vline", help="""
+    Whether to consider hover pointer as point (x/y values), or a
+    span on h or v directions.
+    """)
+
+    hit_value_mode = Enum("snap_to_data", "glyph_center", "hit_interpolate",
```

Comment:
This is really more about the tooltip position, it doesn't actually affect the hit testing. Suggest:
```
tooltip_position = Enum('snap_to_data', 'follow_mouse', 'interpolate')
```
This is still slightly wonky maybe, not all of these apply to all glyphs. e.g., 'interpolate' only has meaning for lines, really. Maybe this would be better:
```
    snap_to_data = Bool()  # False is equiv to "follow_mouse"
    allow_interpolation = Bool()
```
This would convey (to me anyway) that interpolation is a thing that could be permitted, but might not always be applicable. It would also not break the existing interface for `snap_to_data`